### PR TITLE
*: decouple planner/core from ddl package

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary:

Previously, the package `planner/core` depends on the package `ddl`. It is a bit confusing.

### What is changed and how it works?

1. Unify the functions of `model.TableInfo` or `model.IndexInfo`.
2. Decouple planner/core from ddl package.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: types/field_type.go [hunk 1]
```diff
@@ -1448,3 +1449,17 @@ func ConvertBetweenCharAndVarchar(oldCol, newCol byte) bool {
 	return (IsTypeVarchar(oldCol) && newCol == mysql.TypeString) ||
 		(oldCol == mysql.TypeString && IsTypeVarchar(newCol) && collate.NewCollationEnabled())
 }
+
+// IsTooBigFieldLength check if the varchar type column exceeds the maximum length limit.
+func IsTooBigFieldLength(colDefTpFlen int, colDefName, setCharset string) error {
```
Comment (xhebox) [inline types/field_type.go:1454]: I suggest having `varchar` in the function name, since it is used exclusively for varchar.

**Review by xhebox — COMMENTED** [review]
Rest LGTM

**Review by zimulala — APPROVED** [review]
LGTM

File: types/field_type.go (hunk 1 above)
Comment (tangenta) [reply types/field_type.go]: Updated.

**Review by hawkingrei — APPROVED** [review]

**Review by qw4990 — APPROVED** [review]

**Pull request merged**